keps/sig-node/4742-node-topology-downward-api/README.md

```diff
@@ -0,0 +1,409 @@
+* The `PodTopologyLabels` admission plugin is responsible for mutating `pods/binding` subresource, adding topology labels matching the target Node.
+* `PodTopologyLabels` admission will not override any existing topology labels on Pods.
+* A feature gate, `PodTopologyLabelsAdmission` will be introduced in v1.33. Alpha and disabled by default.
+The `PodTopologyLabels` admission plugin can only be set when this feature gate is enabled.
+* The Binding REST implementation will be updated to copy all labels from `pods/binding` subresource into Pods.
+At this point we do not remove labels, it is assumed that any labels in Binding are safe to expose via Downward API.
+* For exposing additional node labels, at the discretion of the cluster admin, a mutating admission webhook can be used to mutate labels of the `pods/binding` subresource.
+
+### Test Plan
+
+[X] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+Unit tests will be added for:
+* New `PodTopologyLabels` admission plugin
+* Binding REST implementation
+
+Unit tests will also ensure behavior is exercised when the feature gate is enabled.
+
+##### Integration tests
+
+Integration tests will be added to test the following behavior:
+* Pods contain topology labels when `PodTopologyLabels` admission plugin is enabled.
+* Topology labels can be expressed in Pod Downward API
+* Node labels outside standard topology labels are disallowed
+* Topology labels are empty if underlying Node does not specify a topology
+
+Integration tests will also ensure behavior is exercised when the feature gate is enabled.
+
+##### e2e tests
+
+E2E tests will be added to test the following scenarios:
+* Pods contain topology labels when `PodTopologyLabels` admission plugin is enabled.
+* A Pod using downward API to retrieve the underlying topology information about the Node
+* A Pod attempting to use downward API to retrieve Node labels that are not the standard topology labels
+* A Pod using downward API on a Node that does not contain any topology information.
+
+E2E tests will also ensure behavior is exercised when the feature gate is enabled.
+
+### Graduation Criteria
+
+#### Alpha
+
+- All standard topology labels can be retrieved using downward API.
+- Behavior is implemented behind a feature gate that is off by default.
+- Initial unit, integration and e2e tests completed and enabled.
+
+#### Beta
+
+TODO after Alpha.
+
+#### GA
+
+TODO after Beta.
+
+### Upgrade / Downgrade Strategy
+
+Upgrade should be relatively safe because there no behavior changes unless topology labels
+are specified in the downward API.
+
+There are some risks to downgrade, for example, if a Pod relies on the topology
+information using downward API, that information could get lost on downgrade.
```

Comment:
please highlight the fact that already running pods will not be affected. It is about the new Pods utilizing this feature

Reply:
done